src/app/styles/app.css

```diff
@@ -79,39 +86,43 @@ body {
 #app {
   display: grid;
   height: 100vh;
-  grid-template-columns: 8em 16em 1fr;
+  grid-template-columns: 8em 16em 1fr 16em;
```

Comment:
I have some reservations about globally adding a 3rd panel/right panel, at least without being able to disable it.

I personally play on a small laptop and generally only want a left-hand side HUD and session manager, and I personally think it would be a mistake to replicate the clutter of the SF UI.

Reply:
I can absolutely see that. Having options seems like a smart way to go, perhaps? I like the idea of having room for more panels, with sacrificing vertical main stream space, but that's just my preference. Customizability here could be a big strength of the client, but still limiting things so it's not messy.

How about a setting for the visibility of both left and right columns? 

`:ui left-panels on/off`
`:ui right-panels on/off`

That could support the idea of someone preferring _only_ right-side placements as well?

Pseudo-code for UI configuration...

```
[{
   panel: "vitals",
   placement: "left"
   order: 1,
   height: "33%"
},
{
   panel: "injuries",
   placement: "left"
   order: 2,
   height: "23%"
},
{
   panel: "chat",
   placement: "right"
   order: 1,
   height: "80%"
}]

Reply:
That seems like a solid middle ground.  Customization is cool, I am just trying to be mindful of the fine line of config hell profanity FE has and maintainability hell that SF has seemed to have had by the lack of updates.

This seems like maybe a good way forward because it's tweakable, but still easily programmatically consumable.

Maybe we can add a `layout` property to the `config.json` and that away you can toggle between layouts if you are working on one more easily.  I think we always got to have sane defaults.

Reply:
What do you think about the chat then?

- Treat it like a Panel that can go in a column?
- Place it at the top of the main feed as the only placement option in an on/off capacity? 
- Create a "top" area that can contain arbitrary Panels also? Or a single Panel?

Reply:
> Place it at the top of the main feed as the only placement option in an on/off capacity?
My preference is probably this, I want room to expand features as a more developed interaction between the scripting engine & FE takes place as well.

I want them to work in unison or at least offer some API the engine can use that's more sane to manipulate the FE.  The whole prevalence of GTK widgets is an example of this problem.  

I want a "fast nav" command bar that integrates with the MapDB and a context-driven UI approach. Less of "I SHOW YOU ALL THE THINGS" and more, "I show you what's important right now"

I'm afraid if we rush to put stuff in the space simply replicating what SF does, we won't be able to back track on that later, and it won't actually bring any advancement over what is currently available.

Reply:
![intelligent-command-bar](https://user-images.githubusercontent.com/1090434/86614605-42c34000-bf81-11ea-8f53-e8686856e864.png)

I want something like this, but for GS

Reply:
OK so I can remove the Panel-ization of Chat here and put it back to an top-placement thing, and this branch can be for the optional sidebars concept.